src/backend/graphItemService.ts

```diff
@@ -0,0 +1,138 @@
+
+import {RestClient} from '@linkurious/rest-client';
+import {GroupedErrors, parseCSV, RowErrorMessage} from './utils';
+import {ImportEdgesParams, ImportItemsResponse, ImportNodesParams} from '../@types/shared';
+
+export class GraphItemService {
+  public static async importGraphItems(
+    params: ImportNodesParams | ImportEdgesParams,
+    rc: RestClient,
+    isEdge: boolean
+  ): Promise<ImportItemsResponse> {
+    const csv = parseCSV(params.csv);
+    const headers = GraphItemService.checkNonEmptyHeaders(csv);
+
+    // A map to group the common errors
+    const errors = new GroupedErrors();
+
+    let i = 0;
+    for (const rowValues of csv) {
+      i++;
+      try {
+        // Merge headers with rowValues
+        const properties = GraphItemService.buildProperties(headers, rowValues, isEdge);
+
+        let response;
+        if (isEdge) {
+          // Import an edge
+          const {sourceKey, itemType, sourceType, destinationType} = params as ImportEdgesParams;
+          response = await rc.graphEdge.createEdge({
+            sourceKey: sourceKey,
+            type: itemType,
+            properties: properties,
+            source: await GraphItemService.getNodeID(rc, sourceKey, sourceType, rowValues[0]),
+            target: await GraphItemService.getNodeID(rc, sourceKey, destinationType, rowValues[1])
+          });
+        } else {
+          // Import a node
+          response = await rc.graphNode.createNode({
+            sourceKey: params.sourceKey,
+            categories: [params.itemType],
+            properties: properties
+          });
+        }
+
+        console.log({response});
+        if (!response.isSuccess()) {
+          errors.add(response, i);
+        }
+      } catch (e) {
+        console.log({e});
+        errors.add(e.message, i);
+      }
+    }
+
+    if (errors.total === 0) {
+      return {
+        success: i
+      };
+    }
+
+    return {
+      success: i - errors.total,
+      failed: errors.total,
+      error: errors.toString()
```

Comment:
Why returning a string here ? it is better to return an object with key 'type of the error' and value the list of the row